### ISSUE TITLE
Oppo Devices - OneSignalSyncServiceUtils syncServicePendingIntent throws Null Pointer Exception

### ISSUE THREAD

**Issue by yaseen-homage** [OneSignal/OneSignal-Android-SDK#491]
```java
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'int com.android.server.job.controllers.JobStatus.getUid()' on a null object reference
       at android.os.Parcel.readException(Parcel.java:1636)
       at android.os.Parcel.readException(Parcel.java:1583)
       at android.app.job.IJobScheduler$Stub$Proxy.schedule(IJobScheduler.java:122)
       at android.app.JobSchedulerImpl.schedule(JobSchedulerImpl.java:42)
       at com.onesignal.OneSignalSyncServiceUtils.syncServicePendingIntent(Unknown Source)
       at com.onesignal.OneSignalSyncServiceUtils.cancelSyncTask(Unknown Source)
       at com.onesignal.OneSignalSyncServiceUtils.scheduleLocationUpdateTask(Unknown Source)
       at com.onesignal.LocationGMS.scheduleUpdate(Unknown Source)
       at com.onesignal.LocationGMS$1.run(Unknown Source)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by jkasten2** [comment]
@yaseen-homage Thanks for reporting, we will look into this and let you know what we find.

**Comment by yaseen-homage** [comment]
@jkasten2  Adding more Info As It might help you to debug
This crash happened few times in past 3 weeks. So far all the crashes are happening only on oppo devices.

DEVICE  : OPPO CPH1607 -OS(6.0), OPPO CPH1611 -OS (6.0.1)

**Comment by jkasten2** [comment]
This stack trace is different but I seems it is related to issue #487

**Comment by jkasten2** [comment]
This should now be fixed in the 3.8.4 update>
https://github.com/OneSignal/OneSignal-Android-SDK/releases/tag/3.8.4
